### PR TITLE
[python] Split Python wheel in two separate wheels

### PR DESCRIPTION
**Pull request by agners**
#### Problem
Currently the Python Controller library is one large wheel, with the native library, the clusters, the interaction model and the REPL packed in a single wheel.

This PR is a first step in break out the library into smaller wheels.

#### Change overview
The first few commits aim to make the current wheel build logic more generic by moving most wheel specific information into `BUILD.gn`. Then this splits it in two wheels: The main wheel named `chip-library` (still fairly monolithic, with cluster library and interaction model in a single wheel), and a `chip-repl`, with the REPL and its dependency.

This lowers the dependency of `chip-library` already. I also plan to split the fairly independent cluster library into a separate wheel in a follow up PR.

#### Testing
Build and runtime tested locally (using `scripts/build_python.sh -m minimal  -i separate` and `chip-repl`).

### CODE REVIEW

File: src/controller/python/BUILD.gn [hunk 1]
```diff
@@ -222,40 +305,69 @@ pw_python_action("python") {
...
+  py_package_name = "chip-library"
+
   public_deps = [ ":ChipDeviceCtrl" ]
 
-  output_name = "chip-0.0-${tags}.whl"
-  outputs = [ "${_dist_dir}/$output_name" ]
+  output_name = "chip_library-${chip_python_version}-${tags}.whl"
+}
+
+chip_python_wheel_action("chip-repl") {
```
Comment (mrjerryjohns) [inline src/controller/python/BUILD.gn:330]: Shouldn't this have a dependency on `chip-library`?

File: BUILD.gn [hunk 2]
```diff
@@ -123,7 +123,7 @@ if (current_toolchain != "${dir_pw_toolchain}/default:default") {
         if (enable_pylib) {
           deps += [ "${chip_root}/src/pybindings/pycontroller" ]
         }
-        deps += [ "${chip_root}/src/controller/python" ]
+        deps += [ "${chip_root}/src/controller/python:chip-repl" ]
```
Comment (mrjerryjohns) [inline BUILD.gn:126]: This won't automatically build `chip-library` too though would it?

File: src/controller/python/BUILD.gn [hunk 3]
```diff
@@ -111,21 +113,79 @@ shared_library("ChipDeviceCtrl") {
...
+        "--lib-name",
+        lib_name,
+      ]
+    }
+
+    outputs = [ "${_dist_dir}/$output_name" ]
+  }
+}
+
+chip_python_wheel_action("chip-library") {
```
Comment (mrjerryjohns) [inline src/controller/python/BUILD.gn:178]: Can we re-name this later to `chip-core`, and `chip-clusters`?

File: BUILD.gn (hunk 2 above)
Comment (agners) [reply BUILD.gn:126]: Right, I had a dependency in the chip-repl target a earlier version, but had to remove it for reasons I don't remember. I'll need to check that again.

File: src/controller/python/BUILD.gn (hunk 3 above)
Comment (agners) [reply src/controller/python/BUILD.gn:178]: Works for me. In a follow up or should I add it to this PR?

File: src/controller/python/BUILD.gn (hunk 1 above)
Comment (agners) [reply src/controller/python/BUILD.gn:330]: Added a runtime dependency using `data_deps`.

**Review by woody-apple — APPROVED** [review]
Fast tracking tooling updates

**Review by mrjerryjohns — APPROVED** [review]

**Pull request merged**